components/eam/src/chemistry/modal_aero/modal_aero_coag.F90

```diff
@@ -1,2337 +1,1363 @@
+      0.97626_wp,0.98254_wp/
+    data (bm3i(10, 8,ibeta ), ibeta=1,10)/   &
+      0.87498_wp,0.88048_wp,0.89264_wp,0.90737_wp,0.92240_wp,0.93642_wp,0.94877_wp,0.95917_wp,   &
+      0.96762_wp,0.97429_wp/
+    data (bm3i(10, 9,ibeta ), ibeta=1,10)/   &
+      0.93946_wp,0.94209_wp,0.94781_wp,0.95452_wp,0.96111_wp,0.96704_wp,0.97203_wp,0.97602_wp,   &
+      0.97900_wp,0.98106_wp/
+    data (bm3i(10,10,ibeta ), ibeta=1,10)/   &
+      0.97977_wp,0.98071_wp,0.98270_wp,0.98492_wp,0.98695_wp,0.98858_wp,0.98970_wp,0.99027_wp,   &
+      0.99026_wp,0.98968_wp/
+
+    ! *** end of data statements *************************************
+
+    !----------------------------------------------------------
+    ! Start calculations
+    !----------------------------------------------------------
+    ! Constants and parameters
+
+    sqrttwo = sqrt(two)
+    dlgsqt2 = one / log( sqrttwo )
+
+    esat01   = exp( 0.125_wp * xxlsgat * xxlsgat )
+    esac01   = exp( 0.125_wp * xxlsgac * xxlsgac )
+
+    esat04  = esat01 ** 4
+    esac04  = esac01 ** 4
+
+    esat05  = esat04 * esat01
+    esac05  = esac04 * esac01
+
+    esat08  = esat04 * esat04
+    esac08  = esac04 * esac04
+
+    esat09  = esat08 * esat01
+    esac09  = esac08 * esac01
+
+    esat16  = esat08 * esat08
+    esac16  = esac08 * esac08
+
+    esat20  = esat16 * esat04
+    esac20  = esac16 * esac04
+
+    esat24  = esat20 * esat04
+    esac24  = esac20 * esac04
+
+    esat25  = esat20 * esat05
+    esac25  = esac20 * esac05
+
+    esat36  = esat20 * esat16
+    esac36  = esac20 * esac16
+
+    esat49  = esat24 * esat25
+
+    esat64  = esat20 * esat20 * esat24
+    esac64  = esac20 * esac20 * esac24
+
+    esat100 = esat64 * esat36
+
+    dgat2   = dgatk * dgatk
+    dgat3   = dgatk * dgatk * dgatk
+    dgac2   = dgacc * dgacc
+    dgac3   = dgacc * dgacc * dgacc
+
+    sqdgat  = sqrt( dgatk )
+    sqdgac  = sqrt( dgacc )
+    sqdgat5 = dgat2 * sqdgat
+    sqdgac5 = dgac2 * sqdgac
+    sqdgat7 = dgat3 * sqdgat
+
+    !------------------------------------------------------------------
+    ! For the free molecular regime:  page h.3 of whitby et al. (1991)
+    !------------------------------------------------------------------
+    r       = sqdgac / sqdgat
+    r2      = r * r
+    r3      = r2 * r
+    rx4     = r2 * r2
+    ri1     = one / r
+    ri2     = one / r2
+    ri3     = one / r3
+    kngat   = two * lamda / dgatk
+    kngac   = two * lamda / dgacc
+
+    ! Calculate ratio of geometric mean diameters
+
+    rat = dgacc / dgatk
+
+    ! Trap subscripts for bm0 and bm0i, between 1 and 10.
+    ! See page h.5 of whitby et al. (1991)
+
+    n2n = max( 1, min( 10,   nint( 4.0_wp * ( sgatk - 0.75_wp ) ) ) )
+    n2a = max( 1, min( 10,   nint( 4.0_wp * ( sgacc - 0.75_wp ) ) ) )
+    n1  = max( 1, min( 10,   1 + nint( dlgsqt2 * log( rat ) ) ) )
+
+    !----------------------------------------------------
```

Comment:
Can we create a new subroutine for the following calculations in "Intermodal coagulation, set up for zeroeth moment"?

Reply:
Done.